### PR TITLE
chore(nuxt): Allow custom middleware handler and options

### PR DESCRIPTION
**Pull request by wobsoriano**
## Description

Previously, Nuxt `clerkMiddleware` was limited to accepting only a custom handler for authentication logic:

```ts
import { clerkMiddleware } from '@clerk/nuxt/server'

export default clerkMiddleware((event) => {
  console.log(event.context.auth)
})
```

There was no way to pass [`authenticateRequest` options](https://clerk.com/docs/references/backend/authenticate-request#authenticate-request-options)

This PR adds support for passing `authenticateRequest` options directly to the middleware, or combining a custom handler with options:

```ts
import { clerkMiddleware } from '@clerk/nuxt/server'

export default clerkMiddleware((event) => {
  console.log(event.context.auth)
}, {
  authorizedParties: ['https://example.com']
})

// or

export default clerkMiddleware({
  authorizedParties: ['https://example.com']
})
```

Just patching for now until we do a GA release

## Checklist

- [x] `pnpm test` runs as expected.
- [x] `pnpm build` runs as expected.
- [x] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [x] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: integration/tests/nuxt/middleware.test.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
+import { expect, test } from '@playwright/test';
+
+import type { Application } from '../../models/application';
+import { appConfigs } from '../../presets';
+import { createTestUtils } from '../../testUtils';
+
+test.describe('custom middleware @nuxt', () => {
```
Comment (LekoArts) [inline integration/tests/nuxt/middleware.test.ts:7]: Let's add a package.json script for this then like:

https://github.com/clerk/javascript/blob/15ce5b110d754442ab92dd2454fbc973c4a66060/package.json#L45

File: packages/nuxt/src/runtime/server/clerkMiddleware.ts
Comment (LekoArts) [inline packages/nuxt/src/runtime/server/clerkMiddleware.ts:1]: Can you add tests for this additional logic? Either as E2E or unit tests 🙏

File: integration/tests/nuxt/middleware.test.ts (hunk 1 above)
Comment (wobsoriano) [reply integration/tests/nuxt/middleware.test.ts:7]: We have one but I'll move it next to vue script to make it easier to see 👍🏼 

https://github.com/clerk/javascript/blob/15ce5b110d754442ab92dd2454fbc973c4a66060/package.json#L39

**Comment by clerk-cookie** [comment]
Hey @wobsoriano - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/astro | 1.5.0-snapshot.v20241126184749 |
| @clerk/chrome-extension | 2.0.1-snapshot.v20241126184749 |
| @clerk/clerk-js | 5.35.1-snapshot.v20241126184749 |
| @clerk/elements | 0.19.8-snapshot.v20241126184749 |
| @clerk/clerk-expo | 2.3.11-snapshot.v20241126184749 |
| @clerk/nextjs | 6.5.1-snapshot.v20241126184749 |
| @clerk/nuxt | 0.0.5-snapshot.v20241126184749 |
| @clerk/clerk-react | 5.17.1-snapshot.v20241126184749 |
| @clerk/remix | 4.2.57-snapshot.v20241126184749 |
| @clerk/tanstack-start | 0.5.4-snapshot.v20241126184749 |
| @clerk/ui | 0.1.25-snapshot.v20241126184749 |
| @clerk/vue | 0.0.9-snapshot.v20241126184749 |

Tip: Use the snippet copy button below to quickly install the required packages.
`@clerk/astro`
```sh
npm i @clerk/astro@1.5.0-snapshot.v20241126184749 --save-exact
```
`@clerk/chrome-extension`
```sh
npm i @clerk/chrome-extension@2.0.1-snapshot.v20241126184749 --save-exact
```
`@clerk/clerk-js`
```sh
npm i @clerk/clerk-js@5.35.1-snapshot.v20241126184749 --save-exact
```
`@clerk/elements`
```sh
npm i @clerk/elements@0.19.8-snapshot.v20241126184749 --save-exact
```
`@clerk/clerk-expo`
```sh
npm i @clerk/clerk-expo@2.3.11-snapshot.v20241126184749 --save-exact
```
`@clerk/nextjs`
```sh
npm i @clerk/nextjs@6.5.1-snapshot.v20241126184749 --save-exact
```
`@clerk/nuxt`
```sh
npm i @clerk/nuxt@0.0.5-snapshot.v20241126184749 --save-exact
```
`@clerk/clerk-react`
```sh
npm i @clerk/clerk-react@5.17.1-snapshot.v20241126184749 --save-exact
```
`@clerk/remix`
```sh
npm i @clerk/remix@4.2.57-snapshot.v20241126184749 --save-exact
```
`@clerk/tanstack-start`
```sh
npm i @clerk/tanstack-start@0.5.4-snapshot.v20241126184749 --save-exact
```
`@clerk/ui`
```sh
npm i @clerk/ui@0.1.25-snapshot.v20241126184749 --save-exact
```
`@clerk/vue`
```sh
npm i @clerk/vue@0.0.9-snapshot.v20241126184749 --save-exact
```

File: packages/nuxt/src/runtime/server/clerkMiddleware.ts
Comment (wobsoriano) [reply packages/nuxt/src/runtime/server/clerkMiddleware.ts:1]: unit test added 🫡

**Review by LekoArts — APPROVED** [review]

**Pull request merged**